documentation/testing_plans.md

```diff
@@ -0,0 +1,815 @@
+adding them to `spec_helper.rb`. You should do this if your module includes
+tests for manifest code, as configuring Puppet and Bolt in `spec_helper.rb` might
+cause your manifest tests to fail.
+
+To include `BoltSpec` helpers and configure Puppet and Bolt for a single test
+file, add the following lines to the file:
+
+```ruby
+# spec/plans/<PLAN NAME>_spec.rb
+
+# Load the spec_helper and BoltSpec library
+require 'spec_helper'
+require 'bolt_spec/plans'
+
+describe '<PLAN NAME>' do
+  # Include the BoltSpec library functions
+  include BoltSpec::Plans
+
+  # Configure Puppet and Bolt before running any tests
+  before(:all) do
+    BoltSpec::Plans.init
+  end
+
+  # Tests . . .
+end
+```
+
+## Running tests
+
+To run tests for your modules, including tests that you write for plans, run the
+following command:
+
+```shell
+bundle exec rake spec
+```
+
+This command runs a rake task that is defined in the `Rakefile` created by PDK,
+and should execute successfully before you have written any tests. After you've
+written tests and run the rake task, the task prints a list of your tests to the
+console together with each test's pass or fail status.
+
+## Configuration
+
+By default, the testing environment uses Bolt's default configuration and an
+empty inventory. The one exception is that Bolt's modulepath is configured to
+the modulepath set up for `rspec-puppet`.
+
+You can create your own values for Bolt's configuration, inventory, and
+modulepath by overriding the helper functions that set them. You can write
+these functions in the `spec_helper.rb` file, which exposes them to all tests,
+or in individual test files.
+
+### Overriding configuration
+
+To override the default configuration used by `BoltSpec`, write a `config_data`
+function in either your `spec_helper.rb` file or in the individual test files.
+This function should return a hash of configuration data that matches the
+structure expected in a [configuration file](bolt_project_reference.md).
+
+For example, the following `config_data` function loads configuration data from
+a test fixture:
+
+```ruby
+def config_data
+  YAML.load_file(File.expand_path('../fixtures/config.yaml'))
+end
+```
+
+### Overriding inventory
+
+To override the default inventory used by `BoltSpec`, write an `inventory_data`
+function. This function should return a hash of inventory data that matches the
+structure expected in an [inventory file](inventory_files.md).
+
+For example, the following `inventory_data` function loads inventory data from a
+test fixture:
+
+```ruby
+def inventory_data
+  YAML.load_file(File.expand_path('../fixtures/inventory.yaml'))
+end
+```
+
+### Overriding the modulepath
+
+To override the default modulepath used by `BoltSpec`, write a `modulepath`
+function. This function should return an array of strings, where each string is
+the path to a directory that includes modules.
+
+```ruby
+def modulepath
+  [File.expand_path('../fixtures/modules')]
+end
+```
+
+## Running plans
```

Comment:
```suggestion
## Acceptance Testing for Bolt Plans
```
Since this actually runs the plan with the provided parameters, I think it's safe to call this acceptance testing.